src/core/esbuild.ts

```diff
@@ -4,8 +4,61 @@
  * Copyright (C) 2021-2022 Posit Software, PBC
  */
 
+import { assert } from "testing/asserts";
 import { execProcess } from "./process.ts";
 import { architectureToolsPath } from "./resources.ts";
+import { TempContext } from "./temp-types.ts";
+import { createTempContext } from "./temp.ts";
+import { kQuartoVersion } from "../config/constants.ts";
+
+type ESBuildAnalysisImport = {
+  path: string;
+  kind: string;
+  external: boolean;
+};
+
+type ESBuildOutputValue = {
+  imports: ESBuildAnalysisImport[];
+  entryPoint: string;
+  inputs: Record<string, { bytesInOutput: number }>;
+  bytes: number;
+};
+
+export type ESBuildAnalysis = {
+  inputs: Record<string, { bytes: number; format: string }>;
+  outputs: Record<string, ESBuildOutputValue>;
+};
+
+export async function esbuildAnalyze(
+  input: string,
+  workingDir: string,
+  tempContext?: TempContext,
+): Promise<ESBuildAnalysis> {
+  let mustCleanup = false;
+  if (!tempContext) {
+    tempContext = createTempContext();
+    mustCleanup = true;
+  }
+
+  try {
+    const tempName = tempContext.createFile({ suffix: ".json" });
+    await esbuildCommand(
+      [
+        "--analyze=verbose",
+        `--metafile=${tempName}`,
+        "--outfile=/dev/null",
```

Comment:
Is this the part to run also on windows you asked me for ? 

Do you want me to check how it works on windows ?

Reply:
Yes, that's it - I still need to make the changes!

Reply:
Fixed by [de6f206](https://github.com/quarto-dev/quarto-cli/pull/11821/commits/de6f2066a8841ed181f4e0589b88bbc583ae2dad)